### PR TITLE
bug fix 770, 771 Save filters and list view after clicking the browser back

### PR DESCRIPTION
**Pull request by yulkats**

### CODE REVIEW

File: src/app/component/eco-news/components/filter-news/filter-news.component.ts [hunk 1]
```diff
@@ -35,14 +35,26 @@ export class FilterNewsComponent implements OnInit {
   public toggleFilter(currentFilter: string): void {
     this.filters.map(el => el.isActive = el.name === currentFilter ? !el.isActive : el.isActive);
     this.emitActiveFilters();
+    this.setSessionStorageFilters();
   }
 
   private setTags(tags: Array<string>): void {
-    this.filters = tags.map((filter: string) => ({name: filter, isActive: false}));
+    const storageFilters = this.getSessionStorageFilters();
+    this.filters = tags.map((filter: string) => ({ name: filter, isActive: storageFilters.find(el => el === filter) !== undefined }));
```
Comment (mplesha) [inline src/app/component/eco-news/components/filter-news/filter-news.component.ts:43]: correct way to test undefined variable is to use `typeof`

File: src/app/component/eco-news/components/news-list/news-list.component.ts [hunk 2]
```diff
@@ -74,9 +76,21 @@ export class NewsListComponent implements OnInit, OnDestroy {
...
+      this.ecoNewsService.getNewsListByTags(this.iterator++, 12, this.gridOutput)
+        .subscribe((list: EcoNewsDto) => this.setList1(list)) :
+      this.ecoNewsService.getEcoNewsListByPage(this.iterator++, 12)
+        .subscribe((list: EcoNewsDto) => this.setList1(list));
   }
 
+  private setList1(data: EcoNewsDto): void {
+    this.remaining = data.totalElements;
+    this.elements = [...this.elements, ...data.page];
+  }
```
Comment (Gusvarov) [inline src/app/component/eco-news/components/news-list/news-list.component.ts:94]: bad method name

File: src/app/component/eco-news/components/news-list/news-list.component.ts [hunk 3]
```diff
@@ -74,9 +76,21 @@ export class NewsListComponent implements OnInit, OnDestroy {
...
-    this.elements = [...this.elements, ...data.page];
+    this.elements = this.scroll ? [...this.elements, ...data.page] : [...data.page];
+  }
+
+private addElemsToCurrentList1(): void {
+    this.ecoNewsSubscription = this.gridOutput ?
+      this.ecoNewsService.getNewsListByTags(this.iterator++, 12, this.gridOutput)
+        .subscribe((list: EcoNewsDto) => this.setList1(list)) :
+      this.ecoNewsService.getEcoNewsListByPage(this.iterator++, 12)
+        .subscribe((list: EcoNewsDto) => this.setList1(list));
```
Comment (Gusvarov) [inline src/app/component/eco-news/components/news-list/news-list.component.ts:87]: Bad method name

**Comment by mplesha** [comment]
conflicted PR

File: src/app/component/eco-news/components/filter-news/filter-news.component.ts (hunk 1 above)
Comment (yulkats) [reply src/app/component/eco-news/components/filter-news/filter-news.component.ts]: fixed

File: src/app/component/eco-news/components/news-list/news-list.component.ts (hunk 2 above)
Comment (yulkats) [reply src/app/component/eco-news/components/news-list/news-list.component.ts]: fixed

File: src/app/component/eco-news/components/news-list/news-list.component.ts (hunk 3 above)
Comment (yulkats) [reply src/app/component/eco-news/components/news-list/news-list.component.ts]: fixed

**Pull request merged**